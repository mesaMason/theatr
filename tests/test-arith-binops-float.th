func main() -> int:
    double f;
    f = 0.5 + 0.5;
    print(f);
    f = .0 + 0.1;
    print(f);
    f = -.9 + 0.1;
    print(f);
    
    f = -.9 - .01;
    print(f);    
    print(0.0 - 1.5);
    
    f = 4.0
    double d;
    d = f * 5.0
    print(d);
    print(5.0 * f);
    print(5.0 * -.2);

    print(4.2 / 1.0);
    print(1.0 / 3.0);
    print(4.0 / -1.0);

    return 0;
