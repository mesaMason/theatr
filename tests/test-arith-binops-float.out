1.000000
0.100000
-0.800000
-0.910000
-1.500000
20.000000
20.000000
-1.000000
4.200000
0.333333
-4.000000
